### PR TITLE
push images from main CI build to ghcr.io

### PR DESCRIPTION
**Pull request by pkwarren**
Update images published from the main CI build to ghcr.io (for future
caching).

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -35,10 +35,10 @@ jobs:
       id: buildx
       uses: docker/setup-buildx-action@v2
       with:
-        driver: docker
+        driver-opts: network=host
     - name: Build
       shell: bash
-      run: make
+      run: make DOCKER_ORG="ghcr.io/bufbuild" DOCKER_BUILD_EXTRA_ARGS="--build-arg BUILDKIT_INLINE_CACHE=1 --push"
```
Comment (pkwarren) [inline .github/workflows/ci.yml:43]: `BUILDKIT_INLINE_CACHE=1` will publish cache information along with images to the container registry.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -4,9 +4,9 @@ on:
     branches: [ "main" ]
   workflow_dispatch: {} # support manual runs
 
-# Prevent writing to the repository using the CI token.
 # Ref: https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#permissions
-permissions: read-all
+permissions:
+  packages: write
```
Comment (pkwarren) [inline .github/workflows/ci.yml:9]: Hope this is all we need.

**Review by mfridman — APPROVED** [review]

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -35,10 +35,12 @@ jobs:
       id: buildx
       uses: docker/setup-buildx-action@v2
       with:
-        driver: docker
+        driver-opts: network=host
+    - name: Log in to registry
+      run: echo "${{ secrets.GITHUB_TOKEN }}" | docker login ghcr.io -u $ --password-stdin
```
Comment (pkwarren) [inline .github/workflows/ci.yml:40]: Took this from https://docs.github.com/en/packages/managing-github-packages-using-github-actions-workflows/publishing-and-installing-a-package-with-github-actions#upgrading-a-workflow-that-accesses-ghcrio.

File: .github/workflows/ci.yml (hunk 2 above)
Comment (mfridman) [reply .github/workflows/ci.yml:9]: Based on the permissions page, I believe so. Let's give it a try.

https://docs.github.com/en/actions/security-guides/automatic-token-authentication#permissions-for-the-github_token

**Pull request merged**